### PR TITLE
Support reading pixels from a web link

### PR DESCRIPTION
**Pull request by shonk-msft**
Support reading pixels from a web link.
Also, add a sample notebook illustrating representation and manipulation of images as tensors.

### CODE REVIEW

File: src/Core/Rexl.Harness/DisposingStream.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    public override void Write(byte[] buffer, int offset, int count) => _inner.Write(buffer, offset, count);
+    public override void Write(ReadOnlySpan<byte> buffer) => _inner.Write(buffer);
+    public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) =>
+        _inner.WriteAsync(buffer, offset, count, cancellationToken);
+    public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        => _inner.WriteAsync(buffer, cancellationToken);
+    public override void WriteByte(byte value) => _inner.WriteByte(value);
+    public override int WriteTimeout { get => _inner.WriteTimeout; set => _inner.WriteTimeout = value; }
+
+    // Not needed. Temporarily set to true then start a line with "override" to get VS to help indicate whether
```
Comment (shonk-msft) [inline src/Core/Rexl.Harness/DisposingStream.cs:92]: >Not [](http://example.com/codeflow?start=7&length=3)

Awkward wording.

File: src/Core/Rexl.Harness/Storage.cs [hunk 2]
```diff
@@ -61,12 +62,16 @@ public virtual Task<(Link full, Stream stream)> LoadStreamAsync(Link linkCtx, Li
...
-    /// from/to the local file system.
+    /// from/to the local file system and http.
     /// </summary>
     public abstract class LocalFileStorage : Storage
     {
+        private readonly HttpClient _http;
+
         protected LocalFileStorage()
         {
+            _http = new HttpClient();
```
Comment (jeburge-msft) [inline src/Core/Rexl.Harness/Storage.cs:73]: Should LocalFileStorage be disposable and dispose of this HttpClient?

File: src/Core/Rexl.Harness/Storage.cs [hunk 3]
```diff
@@ -61,12 +62,16 @@ public virtual Task<(Link full, Stream stream)> LoadStreamAsync(Link linkCtx, Li
...
+            _http.DefaultRequestHeaders.UserAgent.ParseAdd("Mozilla/5.0 (compatible; Rexl/1.0)");
```
Comment (jeburge-msft) [inline src/Core/Rexl.Harness/Storage.cs:74]: I'm assuming this header is only needed for Mozilla integration?

**Review by jeburge-msft — APPROVED** [review]

File: src/Core/Rexl.Harness/Storage.cs (hunk 2 above)
Comment (shonk-msft) [reply src/Core/Rexl.Harness/Storage.cs:73]: Generally, a `LocalFileStorage` is intended to live as long as a `Harness`, which is typically for the lifetime of the host application.

File: src/Core/Rexl.Harness/Storage.cs (hunk 3 above)
Comment (shonk-msft) [reply src/Core/Rexl.Harness/Storage.cs:74]: This is needed for some web servers. For example wikimedia fails if there is no user agent. The sample notebooks added both use images from wikimedia.

**Pull request merged**